### PR TITLE
Update ls.rs

### PR DESCRIPTION
**Pull request by k0pernicus**
2 new options when listing files/directories to get:
- the size of the file/directory,
- the mode of the file/directory.

**TODO** : Improve code (especially to list a single file) and add new options like modified time.

### CODE REVIEW

**Comment by k0pernicus** [comment]
The check failed because I have modified a lib in `redox` to use called functions in `ls.rs`.  
Can I have to wait for the core PR before to update a Redox submodule?

File: src/bin/ls.rs [hunk 1]
```diff
@@ -23,28 +31,57 @@ OPTIONS
...
         let read_dir = Path::new(path).read_dir().try(stderr);
-
         let mut entries = vec![];
         for dir in read_dir {
             let dir = match dir {
                 Ok(x) => x,
                 Err(_) => continue,
             };
-            let mut file_name = dir.file_name().to_string_lossy().into_owned();
+            let mut to_return = "".to_string();
```
Comment (ticki) [inline src/bin/ls.rs]: Use `String::new()` to avoid allocations etc.

File: src/bin/ls.rs [hunk 2]
```diff
@@ -56,26 +93,39 @@ fn list_dir(path: &str, stdout: &mut StdoutLock, stderr: &mut Stderr) {
...
+
 fn main() {
     let stdout = stdout();
     let mut stdout = stdout.lock();
     let mut stderr = stderr();
 
+    let mut path_file = ".".to_string();
+
+    let mut flags_struct = Flags {
+        mode : false,
```
Comment (ticki) [inline src/bin/ls.rs]: `" : "` -> `": "`

File: src/bin/ls.rs [hunk 3]
```diff
@@ -56,26 +93,39 @@ fn list_dir(path: &str, stdout: &mut StdoutLock, stderr: &mut Stderr) {
...
-            exit(0);
+        match arg.as_str() {
+            "-h" | "--help" => {
+                stdout.write(MAN_PAGE.as_bytes()).try(&mut stderr);
+                stdout.flush().try(&mut stderr);
+                exit(0);    
+            },
+            "-m" | "--mode" => flags_struct.mode = true,
+            "-s" | "--size" => flags_struct.size = true,
+            e @ _ => {
```
Comment (ticki) [inline src/bin/ls.rs]: `e @ _` is equivalent to `e`.

**Comment by k0pernicus** [comment]
Thanks @ticki - I will update my code soon :-)

**Comment by jackpot51** [comment]
You should ensure that this will compile on Linux, which means using https://doc.rust-lang.org/std/os/unix/fs/trait.MetadataExt.html

**Comment by jackpot51** [comment]
This has been done a different way

**Pull request closed without merging**